internal/filter/types.go

```diff
@@ -73,116 +112,108 @@ func NewCondition(column string, value string) *Condition {
 	}
 }
 
-type Exists Condition
+// Eval evaluates this Condition based on its operator.
+// Returns true when the filter evaluates to true false otherwise.
+func (c *Condition) Eval(filterable Filterable) (bool, error) {
+	var err error
+	match := false
+	switch c.op {
+	case Equal:
+		match, err = filterable.EvalEqual(c.column, c.value)
+		if err != nil {
+			return false, err
+		}
+	case UnEqual:
+		match, err = filterable.EvalEqual(c.column, c.value)
+		if err != nil {
+			return false, err
+		}
 
-func NewExists(column string) *Exists {
-	return &Exists{column: column}
-}
+		match = filterable.EvalExists(c.column) && !match
```

Comment:
I find it surprising that `EvalEqual` is done before `EvalExists`. But at the very least, it should be `!match && EvalExists()`, no need to run that function if it can't influence the result anyways.